keps/sig-scheduling/5142-pop-backoffq-when-activeq-empty/README.md

```diff
@@ -0,0 +1,409 @@
+- Make sure [backoff in case of error](#backoff-wont-be-working-as-natural-rate-limiter-in-case-of-errors) is not skipped.
+
+#### GA
+
+- Gather feedback from users and fix reported bugs.
+
+### Upgrade / Downgrade Strategy
+
+**Upgrade**
+
+During the alpha period, users have to enable the feature gate `SchedulerPopFromBackoffQ` to opt in this feature.
+This is purely in-memory feature for kube-scheduler, so no special actions are required outside the scheduler.
+
+**Downgrade**
+
+Users need to disable the feature gate.
+
+### Version Skew Strategy
+
+This is purely in-memory feature for kube-scheduler, and hence no version skew strategy.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `SchedulerPopFromBackoffQ`
+  - Components depending on the feature gate: kube-scheduler
+
+###### Does enabling the feature change any default behavior?
+
+Pods that are backing off might be scheduled earlier when activeQ is empty.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes.
+The feature can be disabled in Alpha and Beta versions
+by restarting kube-scheduler with the feature-gate off.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The scheduler again starts to pop pods from backoffQ when activeQ is empty.
+
+###### Are there any tests for feature enablement/disablement?
+
+Given it's purely in-memory feature and enablement/disablement requires restarting the component (to change the value of feature flag), 
+having feature tests is enough.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+The partly failure in the rollout isn't there because the scheduler is the only component to rollout this feature. 
+But, if upgrading the scheduler itself fails somehow, new Pods won't be scheduled anymore,
+while Pods, which are already scheduled, won't be affected in any case.
+
+###### What specific metrics should inform a rollback?
+
+Abnormal values of metrics related to scheduling queue, meaning pods are stuck in activeQ:
+- `scheduler_schedule_attempts_total` metric with `scheduled` label is almost constant, while there are pending pods that should be schedulable. 
+  This could mean that pods from backoffQ are taken instead of those from activeQ.
+- `scheduler_pending_pods` metric with `active` label is not decreasing, while with `backoff` is almost constant.
+- `scheduler_queue_incoming_pods_total` metric with `PopFromBackoffQ` label is increasing when there are pods in activeQ.
+- `scheduler_pod_scheduling_sli_duration_seconds` metric is visibly higher for schedulable pods.
+
```

Comment:
Can we detect bugs potentially by checking if `scheduler_queue_incoming_pods_total` with `PopFromBackoffQ` is _always_ higher than other `event` labels? (i.e., apparently too many pods are moved by this feature always